File: hamster-back/library-core/src/main/java/com/librarycore/book/domain/BookSku.java
package com.librarycore.book.domain;

import identity.BookInventoryId;
import identity.BookSkuId;
import name.Isbn;

import java.util.List;

public record BookSku(
        BookSkuId bookSkuId,
        String title,
        String author,
        Isbn isbn,
        List<BookInventory> inventories

) {
    public void addInventories(int quantity) {
        if (quantity <= 0) throw new IllegalArgumentException("수량은 0보다 커야 합니다.");
        for (int i = 0; i < quantity; i++) {
            this.inventories.add(
                    new BookInventory(
                            BookInventoryId.create(),
                            this.bookSkuId,
                            BookStatus.AVAILABLE
                    )
            );
        }
    }
}

File: hamster-back/library-core/src/main/kotlin/com/librarycore/book/domain/BookSku.kt
package com.librarycore.book.domain

import identity.BookInventoryId
import identity.BookSkuId
import name.Isbn

data class BookSku(
    val bookSkuId: BookSkuId,
    val title: String,
    val author: String,
    val isbn: Isbn,
    val inventories: MutableList<BookInventory>,

) {
    fun addInventories(quantity: Int) {
        require(quantity > 0) { "수량은 0보다 커야 합니다." }
        for (i in 0..<quantity) {
            this.inventories!!.add(
                BookInventory(
                    BookInventoryId.create(),
                    this.bookSkuId,
                    BookStatus.AVAILABLE
                )
            )
        }
    }
}